cpp/src/binaryop/compiled/operation.cuh

```diff
@@ -91,11 +91,33 @@ struct TrueDiv {
 };
 
 struct FloorDiv {
-  template <typename T1, typename T2>
-  __device__ inline auto operator()(T1 const& lhs, T2 const& rhs)
-    -> decltype(floor(static_cast<double>(lhs) / static_cast<double>(rhs)))
+  template <
+    typename TypeLhs,
+    typename TypeRhs,
+    std::enable_if_t<(std::is_integral_v<std::common_type_t<TypeLhs, TypeRhs>> or
+                      (cudf::is_fixed_point<TypeLhs>() and std::is_same_v<TypeLhs, TypeRhs>) or
+                      is_duration<TypeLhs>())>* = nullptr>
+  __device__ inline auto operator()(TypeLhs x, TypeRhs y) -> decltype(x / y)
+  {
+    return x / y;
```

Comment:
This is not right for negative values, since in C++ integer division rounds towards zero, rather than -infinity (as required in Python).

This would do the trick:

```suggestion
    if ((x ^ y) >= 0) {
      return x / y;
    } else {
       auto quot = x / y;
       return quot - !!(x - quot * b);
    }
```

We should then probably narrow the type-overload to only do this in the case of signed types (although GCC can optimise to `x / y` in the case that both types are unsigned (for which `(x^y) >= 0` is always true)).

Question: is floor division like this for division of integer columns actually what is wanted in SPARK/Java. Java at least has the same semantics as C++ (round to zero)